docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,598 @@
+
+For interpreting a fetch into a `Future` we must first import the `MonadError[Future, Throwable]` available in cats.
+
+```scala
+import cats.std.future._
+```
+
+We can now interpret a fetch into a future:
+
+```scala
+import cats.std.future._
+
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+val fch: Fetch[(User, Post)] = (getUser(1) |@| getPost(1)).tupled
+
+val fut: Future[(User, Post)] = Fetch.run[Future](fch)
+// Fetching users List(1)
+// Fetching posts List(1)
+
+Await.result(fut, 1 seconds)
+//=> (User, Post) = (User(1,@egg_1),Post(1,2,An article with id 1))
+```
+
+# Caching
+
+As we have learned, Fetch caches intermediate results implicitly using a cache. You can
+provide a prepopulated cache for running a fetch, replay a fetch with the cache of a previous
+one and even implement a custom cache.
+
+## Prepopulating a cache
+
+We'll be using the default in-memory cache, prepopulated with some data. The cache key of an identity
+is calculated with the `DataSource`'s `identity` method; we can pass a cache as the second argument when
+running a fetch with `Fetch.run`.
+
+```scala
+val fch: Fetch[User] = getUser(1)
+val cache = InMemoryCache(UserSource.identity(1) -> User(1, "@dialelo"))
+
+val result: User = Fetch.run[Id](fch, cache)
+//=> result: User = User(1,@dialelo)
+```
+
+As you can see, when all the data is cached no fetch is executed at all since the results are available
+in the cache.
+
+If only part of the data is cached, the cached data won't be asked for:
+
+```scala
+val fch: Fetch[List[User]] = List(1, 2, 3).traverse(getUser)
+val cache = InMemoryCache(UserSource.identity(2) -> User(2, "@two"))
+
+val result: List[User] = Fetch.run[Id](fch, cache)
+// Fetching users List(1, 3)
+
+//=> result: List[User] = List(User(1,@egg_1), User(2,@two), User(3,@egg_3))
+```
+
+## Replaying a fetch
+
+When running a fetch we are generally interested in its final result. However, we also have access to the cache
+and information about the executed rounds once we run a fetch. Fetch's interpreter keeps its state in an environment
+(implementing the `Env` trait) and we can get both the environment and result after running a fetch using `Fetch.runFetch`
+instead of `Fetch.run`.
+
+Knowing this, we can replay a fetch reusing the cache of a previous one. The replayed fetch won't have to call any of the
+data sources.
+
+```scala
+val fch: Fetch[List[User]] = List(1, 2, 3).traverse(getUser)
+
+val env: Env = Fetch.runEnv[Id](fch)
+// Fetching users List(1, 2, 3)
+
+val result: List[User] = Fetch.run[Id](fch, env.cache)
+//=> result: List[User] = List(User(1,@egg_1), User(2,@egg_2), User(3,@egg_3))
+```
+
+## Implementing a custom cache
+
+The default cache is implemented as an in-memory map, but users are free to use their own caches when running a fetch. Your cache should implement the `DataSourceCache` trait, and after that you can pass it to Fetch's `run` methods.
```

Comment:
Perhaps worth mentioning here that the cache is trampolined inside a state monad when evaluating a Fetch so people understand they can easily implement caches with inmutable data structures and not mutable maps.